### PR TITLE
airbyte-ci: disable test reports PR comments

### LINKED ISSUE #29300: Update Connector CI to report status to Github Status Checks not comments
Some connector developers are wary of pushing often to Github because CI spams comments onto their PRs, especially if you are working on multiple connectors.  This makes seeing PR reviews and comments hard.

Right now, CI leaves both a comment and a Github Status check (1,2) for every test (from https://github.com/airbytehq/airbyte/pull/29213) 

Screenshot 2023-08-09 at 12 13 17 PM
Screenshot 2023-08-09 at 12 13 13 PM

Now that we have excellent CI reports ([e.g.](https://ff2e9b0faf2538f560688912502dce1da66d5d28ab8531a0b70ed7f-apidata.googleusercontent.com/download/storage/v1/b/airbyte-ci-reports-multi/o/airbyte-ci%2Fconnectors%2Ftest%2Fpull_request%2Fedgao_faker_schema%2F1691538991%2Fc1e8cf503fbc8710da43995f83f6162815599395%2Fsource-faker%2F5.0.0%2Foutput.html?jk=ATmLuOH1vZ_uWOTfKiL4wULjpiMYD283YR3L99ZlGOP1I85v_Em6-GIreBi-xShiG37RnokCGjfw3pS2je_noyWr7a1YzSwmaio0LV6As6Rb8omtwKc_oApLU6YpGLwb_bwDII8ihFijPrIuKmGwi96x8tdXp6RXrg7Lf4ngQ_dhaAKek8sDkeoO_pcA_7YpgSuJci9pKAVcP42xSUrSjxunuBDQEDN_C_BkG4FeBCpbQRTy_-n4tfTipv0ArOE3qE3v6lHagVOqaYsbTVowZQxIxLs4ic7y9MpdFJJIiu3MwawVAzCshur19p9JPnt8hmT_lD5APP5QsysQxgUWeI4MxJYYYuS1EhW37lBRkQkiKzkErmLW4_VOvEtLHdmg9SuoFJKw5Y3zyuZBrGfQdlp1q6Ywh99jfYMZKV1eeqQ3t66zsWQ1Fpfh_WWilexKLxaiHOGD5-HPKJTP4mCMdDeypCdSTRCfPqHuivBnn21r6afifYlLD2WVztCxBMiF1v5riUmTvW24rtfDIO8Hg7AnjozE-gRVdghgG8tk1__x13a5MlB_eSIjR3dYYzHY_hRFKgPwi7NfsyyJTFT7nazEplOtOzJQUiufPtUkSCDj5T8cd9dXIZWSHEmu_ckfeCKotgQRnrnFO3r-BgipNYWSN3EIEUBh6mmf2TzjXFgt0apurd-_esy_1CCVq6PtxCr9UVVTYBPQL2fBlIwxImrdpSE19ntxKTHKJdtsllZzfuyNwhz562dpOkgMUQUqXKz7CEA9b99ywBE1hIARJoAb48iOwrQbgzeGC-sXUqcU6tOqf4yiikdGcQexieEBLEulV7uieWakOhfTo-Fn8olsbvGzZREyhlpGFCis2229-BV7rK78g7dNsdezoWAJmNRhJkzabCc1U01N8VahnMHsSKWuWnapN1icU2xz6wj-MR9gcKZAQlLB47ZrskzWQxXq5qRbY3Hghb319xJ0PwYhyvKjqPex1ilMRwhpiexCbCOMh-OzC-5cL1LywBwlgWsJYW-01anl91ii-7_v8cc4hmRTXWAgtEJXg74sF5vj0Sl1dsAXY_4UYiIPsRbQqdxzUQwxiydfTvMnn7gHGvKYQDpmxj4oszN5tegdzMnyFvI1c6tQWZc5-3-TpED8bPp6X_88gtFO8nZAqPr177ZnjpIBOQhaCospc5RiZce4GM0vNbZWh9qLuv768_YUfm_rsQ7UZAWXhfXSsjET4aNn8GSu47UCvzGQq3kYCZBM8-VFW7P7b2t87-3kbOUP2s2BfWX07csz2FMCARrGgdeWEIcujThNHUMFaJ_XvffOcX3rxwXRpbxhbxkLAIRqBxKppychPjQsrhFBXrn9gW-AYMEHN97vjBkMwNibKcOgsGMNmyqJZQx_32priXpSKxeGB6OT&isca=1)), I believe that the need for the Github comment is less than it used to be.  We should change the URL in the Github status to point to this report, rather than the action.  This gives the connector developer the information they need to investigate a failure, without clogging up the PR history.

If we cannot change the status report URL, then we should be updating the existing CI comment rather than making a new one.  This would serve the same goal of spam reduction.

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes #29300
We don't want PR comments anymore on connector CI test run.

## How
- Add a 'feature flag' disabling PR comments
- Change the status *details* link to the HTML report when its ready 

The global Connector CI status details will link to the GHA workflow.
The per connector status details will link to the GHA workflow while they're running and to the HTML report when they're done.

### Why a 'feature flag' instead of deleting the commenting logic
When we will consider running the CI on community PRs I think having these comments back can be a good way to expose high-level results to the community contributors. HTML reports are currently private to not overly expose logs.

### CODE REVIEW

**Comment by alafanechere** [comment]
Current dependencies on/for this PR:
* master
  * **PR #31871** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/31871" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/31871?utm_source=stack-comment).

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/reports.py [hunk 1]
```diff
@@ -25,6 +25,8 @@
 class ConnectorReport(Report):
     """A dataclass to build connector test reports to share pipelines executions results with the user."""
 
+    ENABLE_PR_COMMENTS: bool = False
```
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/reports.py:28]: Im a bit apprehensive of leaving unused code in the code base.

Code is cattle not a pet ([ref](https://www.hava.io/blog/cattle-vs-pets-devops-explained))

I would argue that we remove it, unless theres a reason to keep it around that I may be missing.

**Review by bnchrch — APPROVED** [review]
Just one comment around deadcode

But this is at a spot where im fine to pre-approve!

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/reports.py (hunk 1 above)
Comment (alafanechere) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/reports.py:28]: When we will consider running the CI on community PRs I think having these comments back can be a good way to expose high-level results to the community contributors. HTML reports are currently private to not overly expose logs.

**Pull request merged**